### PR TITLE
tests and docs for cashay-loader

### PR DESCRIPTION
**Pull request by jordanh**
Now it's official! 💯

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,6 +110,7 @@ const store = createStore(rootReducer, {});
 Cashay is front-end agnostic, so instead of passing it through React context or making you replace `react-redux` with something non-vanilla, you can just easily export your singleton from where you created it.
 ```js
 import clientSchema from './clientSchema.json';
+// or use webpack: const clientSchema = require('cashay!../server/utils/getCashaySchema.babel.js');
```
Comment (jordanh) [inline README.md]: No actual changes were made to README.md below this line.

File: src/schema/__tests__/cashay-loader-tests.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import 'babel-register';
+import test from 'ava';
```
Comment (jordanh) [inline src/schema/__tests__/cashay-loader-tests.js]: Hey! I like this ava stuff.

File: README.md [hunk 3]
```diff
@@ -46,11 +53,48 @@ export default () => {
...
+import rootSchema from '../graphql/rootSchema';
+import {transformSchema} from 'cashay';
+import rethinkExit from './rethinkExit';
+
+// side-step 'Error: Schema must be an instance of GraphQLSchema.'
+const graphql = require('graphql').graphql;
+
+export default async () => {
+  const schema = await transformSchema(rootSchema, graphql);
+  rethinkExit();
```
Comment (mattkrick) [inline README.md]: should we just inline this?

**Comment by mattkrick** [comment]
Call me crazy, but i think we should word it so the webpack method is the default & the CLI is the back-up, almost footnote-ish ("opinionated software" and whatnot :wink:). I don't really see them as equals, since if you take the webpack route you'll never have to worry about a stale schema. That alone would be enough to make me use webpack in my project.

**Comment by mattkrick** [comment]
i'll merge for now & we can wordsmith later.

**Comment by simenbrekken** [comment]
I'm getting `Module build failed: ReferenceError: regeneratorRuntime is not defined` when implementing the loader from the docs, and as far as I can see `regeneratorRuntime` isn't actually included anywhere in `lib`.

**Comment by mattkrick** [comment]
hmmm, must be missing a `babel-polyfill` somewhere

**Comment by simenbrekken** [comment]
@jordanh I can't get the loader working after adding `babel-polyfill` to `cashay/src/index.js`

```
import { transformSchema } from 'cashay'
import schema from './index'

// Side-step 'Error: Schema must be an instance of GraphQLSchema.'
const graphql = require('graphql').graphql

export default async () => await transformSchema(schema, graphql)
```

It just keeps complaining `Module build failed: TypeError: doc.then is not a function`. I'm a bit confused since the `async/await` should return a promise no matter what. Could it be something related to the `babel-polyfill`?

**Comment by mattkrick** [comment]
can confirm, `var doc = this.exec(content, this.resource);` is returned an object, not a promise, which isn't supposed to happen...

**Comment by jordanh** [comment]
This is my `getCashaySchema.babel.js`:

```js
require('babel-register');
require('babel-polyfill');
module.exports = require('./getCashaySchema')();
```

**Comment by jordanh** [comment]
@simenbrekken: @mattkrick and I talk about this this morning, I clarified the docs and example for this.

See #58 !

**Pull request merged**